src/main/java/com/gotocompany/depot/message/proto/ProtoLogicalValue.java

```diff
@@ -0,0 +1,90 @@
+package com.gotocompany.depot.message.proto;
+
+import com.google.protobuf.InvalidProtocolBufferException;
+import com.google.protobuf.Message;
+import com.google.protobuf.Struct;
+import com.google.protobuf.Timestamp;
+import com.google.protobuf.Value;
+import com.gotocompany.depot.exception.DeserializerException;
+import com.gotocompany.depot.message.LogicalValue;
+import com.gotocompany.depot.schema.LogicalType;
+import com.gotocompany.depot.schema.proto.ProtoSchema;
+import org.json.JSONObject;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class ProtoLogicalValue implements LogicalValue {
+    private final Message message;
+
+    public ProtoLogicalValue(Message message) {
+        this.message = message;
+    }
+
+    @Override
+    public LogicalType getType() {
+        return new ProtoSchema(message.getDescriptorForType()).logicalType();
```

Comment:
should we save the ProtoSchema object? Why create a new one everytime?

Reply:
Yes. started storing it as class field.